packages/backend/src/controllers/bills/bills.controller.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  Controller,
+  Post,
+  Get,
+  Put,
+  Param,
+  Delete,
+  Body,
+  HttpException,
+  HttpStatus,
+  HttpCode,
+} from '@nestjs/common';
+import {
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiNoContentResponse,
+  ApiOperation,
+  ApiTags,
+  ApiForbiddenResponse,
+  ApiBadRequestResponse,
+} from '@nestjs/swagger';
+import { DecodedIdToken } from 'firebase-admin/auth';
+import { BillModel } from 'src/db/bill/bill.schema';
+import { BillStoreService } from 'src/db/bill/billStore.service';
+import { UserStoreService } from 'src/db/user/userStore.service';
+import { User } from 'src/util/user.decorator';
+import { CreateBillDto } from './dto/create-bill.dto';
+import { BillResponseDto, BillsResponseDto } from './dto/bill-response.dto';
+import { UpdateBillDto } from './dto/update-bill.dto';
+import { PayBillDto } from './dto/pay-bill.dto';
+import { BillUtil } from './bills.util';
+import { Auth } from '../../util/auth.decorator';
+
+@ApiTags('bills')
+@Controller('/api/v1/house/bills')
+@Auth()
+export class BillController {
+  constructor(
+    private readonly billStoreService: BillStoreService,
+    private readonly userStoreService: UserStoreService,
+    private readonly billUtil: BillUtil,
+  ) {}
+
+  @Get()
+  @ApiOperation({ summary: 'Retrieve all bills in the users flat.' })
+  @ApiOkResponse({
+    description: 'Bills retrieved successfully',
+    type: BillsResponseDto,
+  })
+  @ApiBadRequestResponse({ description: 'User is not in a house' })
+  async getBills(@User() user: DecodedIdToken): Promise<BillsResponseDto> {
+    const houseID = (await this.userStoreService.findOneByFirebaseId(user.uid))
+      .house;
+
+    if (!houseID) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    return {
+      bills: await Promise.all(
+        (
+          await this.billStoreService.findAllForHouse(houseID)
+        ).map((bill) => {
+          return this.billUtil.covertBillDocumentToResponseDTO(
+            bill,
+            this.userStoreService,
+          );
+        }),
+      ),
+    };
+  }
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new bill resource' })
+  @ApiCreatedResponse({
+    description: 'Bill created successfully',
+    type: BillResponseDto,
+  })
+  async createBill(
+    @Body() createBillDto: CreateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const owner = await this.userStoreService.findOneByFirebaseId(user.uid);
+    const users = await Promise.all(
+      createBillDto.users.map(async (u) => {
+        return {
+          id: (await this.userStoreService.findOneByFirebaseId(user.uid))._id,
```

Comment:
I may be misunderstanding this but is this meant to be `u.uid` instead? Otherwise we can use `owner._id` instead of having to fetch again?

Reply:
Good catch